source/core/shape.cc

```diff
@@ -494,9 +563,9 @@ double
 CompositeShape<dim>::displaced_volume(const double fluid_density)
 {
   double solid_volume = 0;
-  for (const std::shared_ptr<Shape<dim>> &elem : components)
+  for (auto const &[component_id, component] : components)
     {
-      solid_volume += elem->displaced_volume(fluid_density);
+      solid_volume += component->displaced_volume(fluid_density);
```

Comment:
Just curious, when there is an operation, how is he solid volume not the sum of the 2 shapes and only the volume of the remaining solid?

Reply:
It's because it's an approximate volume, but as it's a function that it never called for non spheres I'll remove it. That way it isn't misleading.